Example/Example/EditorDemoController.swift

```diff
@@ -1032,9 +1032,48 @@ extension EditorDemoController {
 
 extension EditorDemoController: TextViewAttachmentDelegate {
 
+    func exportPreviewImageForVideo(atURL url: URL, onCompletion: @escaping (UIImage) -> (), onError: @escaping () -> ()) {
+        DispatchQueue.global(qos: .background).async {
+            let asset = AVURLAsset(url: url)
+            guard asset.isExportable else {
+                onError()
+                return
+            }
+            let generator = AVAssetImageGenerator(asset: asset)
+            generator.appliesPreferredTrackTransform = true
+            generator.generateCGImagesAsynchronously(forTimes: [NSValue(time: CMTimeMake(2, 1))],
+                                                     completionHandler: { (time, cgImage, actualTime, result, error) in
+                                                        guard let cgImage = cgImage else {
+                                                            onError()
+                                                            return
+                                                        }
+                                                        let image = UIImage(cgImage: cgImage)
+                                                        onCompletion(image)
+            })
+        }
+    }
+
     func textView(_ textView: TextView, attachment: NSTextAttachment, imageAt url: URL, onSuccess success: @escaping (UIImage) -> Void, onFailure failure: @escaping () -> Void) {
+        var urlToRequest = url
```

Comment:
Let's try to see how we can make this method more readable.  Right now it has a lot of detail in it.

For instance:
- Line [1076 onward](https://github.com/wordpress-mobile/AztecEditor-iOS/pull/999/files#diff-e7e4649f5f7465f68641a838cc55412fR1076) could go into a method of its own.
- We could use a guard in [line 1059](https://github.com/wordpress-mobile/AztecEditor-iOS/pull/999/files#diff-e7e4649f5f7465f68641a838cc55412fR1059) since the else condition returns.

I know part of this code was not touched by this PR, but since we're adding code to an already large method, I think we should ensure its readability.